### PR TITLE
MUMUP-2196: Redesign search results

### PR DESCRIPTION
**Pull request by jhanstra**
Search results a la @keirserrie mockups. Clean, simple, googley.

Holding off on Directory and Wisc.edu tabs until we have those in place. With this PR, new search results page is shippable. 

Also note this is a new and separate page from the Browse page. The Browse page will stand as is until we decide what to do with it.

Search: 'email'
image

Search: 'time'
image

Search: 'sched'
image

Search in action (it's blazingly fast :fire:):
screen recording 2015-12-07 at 03 07 pm

### CODE REVIEW

**Comment by keirserrie** [comment]
YES :+1:

File: angularjs-portal-home/src/main/webapp/my-app/marketplace/partials/search-results.html [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  </div>
+  
+  <div 
+    ng-show="portlets.length > 0 && (portlets | filter:searchTermFilter | showApplicable:showAll | showCategory:categoryToShow).length == 0" 
+    class="no-results">
+    <p>No matching MyUW content.</p>
+    <p>Maybe try:</p>
+    <ul>
+      <li ng-show="(portlets | filter:searchTermFilter | showApplicable:showAll).length > 0">
+        <a href="#" ng-click="selectFilter('az','')">Show matching content beyond the &quot;{{categoryToShow}}&quot; category</a>.</li>
```
Comment (apetro) [inline angularjs-portal-home/src/main/webapp/my-app/marketplace/partials/search-results.html]: All the references to `categoryToShow` could be washed away.

File: angularjs-portal-home/src/main/webapp/my-app/marketplace/partials/search-results.html [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+				<a>Directory (# results)</a>
+			</li>
+			<li>
+				<a>Wisc.edu (# results)</a>
+			</li> -->
+		</ul>
+	</div>
+
+
+  <!-- <loading-gif data-object='portlets'></loading-gif> -->
```
Comment (timlevett) [inline angularjs-portal-home/src/main/webapp/my-app/marketplace/partials/search-results.html]: The initial load will still be the 7 seconds in prod (or whatever it is). Do we want this loading state?

File: angularjs-portal-home/src/main/webapp/my-app/main.js [hunk 3]
```diff
@@ -53,7 +53,7 @@ define([
         $routeProvider.
             when('/apps', marketplaceRoutes.main).
```
Comment (timlevett) [inline angularjs-portal-home/src/main/webapp/my-app/main.js]: More of a question for @keirserrie , what do we want to do with the old browse page?

File: angularjs-portal-home/src/main/webapp/my-app/marketplace/partials/search-results.html (hunk 2 above)
Comment (apetro) [inline angularjs-portal-home/src/main/webapp/my-app/marketplace/partials/search-results.html]: Yes. Definitely.  Even if the expected load time were shorter, should code defensively for cases where e.g. the client-side front end is cached but the latest JSON is loading down over a slooooow connection.

**Comment by timlevett** [comment]
:+1:

**Comment by vertein** [comment]
:+1: nicely done

**Pull request merged**